internal/engine/docker_pruner.go

```diff
@@ -0,0 +1,81 @@
+package engine
+
+import (
+	"context"
+	"time"
+
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/pkg/logger"
+)
+
+// How often to prune Docker images while Tilt is running
+const dockerPruneInterval = time.Hour * 12
+
+// Prune Docker objects older than this
+// TODO: configurable from Tiltfile for special cases
+const dockerUntilInterval = time.Hour * 6
+
+type DockerPruner struct {
```

Comment:
let's move new subscribers into sub-packages of engine (see other examples in this directory). Trying to reduce circular deps